target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-whw01-v1.dts

```diff
@@ -0,0 +1,237 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "qcom-ipq4019-ap.dk01.1.dtsi"
+#include <dt-bindings/leds/common.h>
+#include <dt-bindings/gpio/gpio.h>
+
+/ {
+	model = "Linksys WHW01 v1";
+	compatible = "linksys,whw01-v1";
+
+	aliases {
+		led-boot = &led_system_blue;
+		led-running = &led_system_blue;
+	};
+
+	chosen {
+		bootargs-append = " root=/dev/ubiblock0_0";
+	};
+
+	reserved-memory {
+		rsvd1@87000000 {
+			reg = <0x87000000 0x500000>;
+			no-map;
+		};
+
+		wifi_dump@87500000 {
+			reg = <0x87500000 0x600000>;
+			no-map;
+		};
+
+		rsvd2@87B00000 {
+			reg = <0x87b00000 0x300000>;
+			no-map;
+		};
+	};
+
+	soc {
+		gpio_keys {
+			pinctrl-0 = <0x52>;
+			pinctrl-names = "default";
+			compatible = "gpio-keys";
+
+			button@1 {
+				label = "reset";
+				linux,code = <0x198>;
+				gpios = <&tlmm 0x3f 0x01>;
+				linux,input-type = <0x01>;
+			};
+		};
+	};
+};
+
+&tlmm {
+	i2c_0_pins: i2c_0_pinmux {
+		mux {
+			function = "blsp_i2c0";
+			pins = "gpio58", "gpio59";
+			bias-disable;
+		};
+	};
+
+	reset_pinmux {
+		linux,phandle = <0x52>;
+		phandle = <0x52>;
+
+		mux {
+			pins = "gpio63";
+			bias-pull-up;
+		};
+	};
+};
+
+&spi_0_pins {
+	pinmux_cs {
+		pins = "gpio54", "gpio4";
+	};
+	pinconf_cs {
+		pins = "gpio54", "gpio4";
+	};
+};
+
+&blsp1_spi1 {
+	status = "okay";
+	pinctrl-0 = <&spi_0_pins>;
+	pinctrl-names = "default";
+	cs-gpios = <&tlmm 54 GPIO_ACTIVE_HIGH>, <&tlmm 4 GPIO_ACTIVE_HIGH>;
+
+	nor@0 {
+		spi-max-frequency = <0x16e3600>;
```

Comment:
really?

Reply:
Appologies, these nodes were originally taken from the device-tree sliced out of the factory update image and run through `dtc` as a result a lot of the raw values are in the wrong base or without macros. I'll fix these up now.